pkg/config/workloads/workloads.go

```diff
@@ -0,0 +1,139 @@
+	// The value of the map is the default value for that resource.
+	// If a container is configured to use this workload, and does not specify
+	// the annotation with the resource and value, the default value will apply.
+	// Default values do not need to be specified.
+	Resources map[string]string `toml:"resources"`
+}
+
+func (w Workloads) Validate() error {
+	for workload, config := range w {
+		if err := config.Validate(workload); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (w *WorkloadConfig) Validate(workloadName string) error {
+	if w.Label == "" {
+		return errors.Errorf("label shouldn't be empty for workload %q", workloadName)
+	}
+	for resource, defaultValue := range w.Resources {
+		validator, ok := resourcesToDefaultValidator[resource]
+		if !ok {
+			return errors.Errorf("process resource %s for workload %s: resource not supported", resource, workloadName)
+		}
+		if err := validator(defaultValue); err != nil {
+			return errors.Wrapf(err, "process resource %s for workload: default value %s invalid", resource, workloadName, defaultValue)
+		}
+	}
+	return nil
+}
+
+var resourcesToDefaultValidator = map[string]func(string) error{
+	CPUShareResource: cpuShareDefaultValidator,
+	CPUSetResource:   cpusetDefaultValidator,
+}
+
+func cpusetDefaultValidator(set string) error {
+	if set == "" {
+		return nil
+	}
+	_, err := cpuset.Parse(set)
+	return err
+}
+
+func cpuShareDefaultValidator(cpuShare string) error {
+	if cpuShare == "" {
+		return nil
+	}
+	if _, err := resource.ParseQuantity(cpuShare); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (w Workloads) MutateSpecGivenAnnotations(ctrName string, specgen *generate.Generator, sboxAnnotations map[string]string) error {
+	var workload *WorkloadConfig
+	for _, wc := range w {
+		for label, _ := range sboxAnnotations {
+			if wc.Label == label {
+				workload = wc
+				break
+			}
+		}
+	}
+	if workload == nil {
+		return nil
+	}
+	for label, value := range sboxAnnotations {
+		for resource, defaultValue := range workload.Resources {
+			if label != workload.AnnotationPrefix+"."+resource+"/"+ctrName {
+				continue
+			}
+			specMutator, ok := resourcesToSpecMutator[resource]
+			if !ok {
+				// CRI-O bug
+				panic(errors.Errorf("resource %s is not defined", resource))
+			}
+			if value == "" {
+				if defaultValue == "" {
+					return errors.Errorf("unable to mutate spec: no value given for resource %s for workload %s", resource, workload.Label)
+				}
+				value = defaultValue
+			} else {
+			}
+			if err := specMutator(specgen, value); err != nil {
+				return errors.Wrapf(err, "mutating spec given workload %s", workload.Label)
+			}
+		}
+	}
+	return nil
+}
+
+var resourcesToSpecMutator = map[string]func(*generate.Generator, string) error{
+	CPUShareResource: cpuShareSpecMutator,
+	CPUSetResource:   cpusetSpecMutator,
+}
+
+func cpuShareSpecMutator(specgen *generate.Generator, configuredValue string) error {
```

Comment:
what configured value do you expect here?
Will it work with something like:
```
cpu.workload.openshift.io/{container-name}: "400m"
```
?

Reply:
Because under the validation do you expect Quantity and looks like do you expect string with uint?